src/marqo/tensor_search/tensor_search.py

```diff
@@ -1559,7 +1560,19 @@ def get_query_vectors_from_jobs(
             ]
             # TODO how doe we ensure order?
             weighted_vectors = [np.asarray(vec) * weight for vec, weight, content in vectorised_ordered_queries]
-            merged_vector = np.mean(weighted_vectors, axis=0)
+            
+            custom_tensors = q.get_context_tensor() 
+            if custom_tensors is not None:
+                weighted_vectors += [np.asarray(v.vector) * v.weight for v in custom_tensors]
+            
+            try:
+                merged_vector = np.mean(weighted_vectors, axis=0)
+            except ValueError as e:
+                raise errors.InvalidArgError(f"The provided vectors are not in the same dimension of the index."
+                                             f"This causes the error when we do `numpy.mean()` over all the vectors.\n"
```

Comment:
let's leave out `numpy` as this is a Marqo implementation detail (just `mean` is fine).

Reply:
This is a copy paste. I want to minimise refactors during a code reshuffle.